src/MilleSabordGameBoard.js

```diff
@@ -1,152 +1,152 @@
+/* eslint-disable import/max-dependencies */
 import React from "react"
 
-import { DiceSet } from "./DiceSet"
-import { rollDice, removeFromArray } from "./helpers/DiceHelpers.js"
-import { getMixedDeck } from "./helpers/CardsHelpers"
-import {
-  computeScore,
-  removeSkullsFromArray,
-  countSymbolsOccurences,
-  isGameOver,
-} from "./helpers/ScoreHelpers"
+import { DiceSet } from "./Dice/DiceSet.js"
+import { ButtonRoll } from "./Dice/ButtonRoll.js"
+import { CurrentRoundScore } from "./Score/CurrentRoundScore.jsx"
+import { TotalScore } from "./Score/TotalScore.jsx"
+import { CardArea } from "./Cards/CardArea.js"
+import { ButtonRestart } from "./ButtonRestart.js"
+import { getMixedDeck } from "./Cards/CardsHelpers.js"
+import { computeScore, isGameOver } from "./Score/ScoreHelpers.js"
+import { DICE_ARRAY, rollOnGoingDices } from "/src/Dice/DiceHelpers.js"
+import { SYMBOL_SKULL } from "/src/symbols/symbol-types.js"
+import { CARD_WITCH } from "src/Cards/card-types.js"
 
 export const MilleSabordGameBoard = () => {
-  const [diceRolled, setDiceRolled] = React.useState([])
+  const [diceOnGoing, setDiceOngoing] = React.useState(DICE_ARRAY)
   const [diceKept, setDiceKept] = React.useState([])
 
   const [totalScore, setTotalScore] = React.useState(0)
 
   const [cardDeck, setCardDeck] = React.useState(getMixedDeck())
   const [currentCard, setCurrentCard] = React.useState({})
 
-  const [roundStarted, setRoundStarted] = React.useState(false)
+  const [diceRolledOnce, setDiceRolledOnce] = React.useState(false)
   const [roundFinished, setRoundFinished] = React.useState(false)
+  const [cardDrawn, setCardDrawn] = React.useState(false)
 
   const clearDiceSet = () => {
-    setDiceRolled([])
+    setDiceOngoing([...diceOnGoing, ...diceKept])
     setDiceKept([])
-    setRoundStarted(false)
+    setDiceRolledOnce(false)
     setRoundFinished(false)
   }
 
   const rollTheDice = () => {
-    const numberOfDice = diceRolled.length
-    const roll = rollDice(numberOfDice > 0 ? numberOfDice : 8)
-
-    // if the roll contain some skulls, remove them from the current roll, add them to the kept dice
-    const numberOfSkulls = countSymbolsOccurences(roll).skull
-    if (numberOfSkulls) {
-      for (var i = 0; i < numberOfSkulls; i++) diceKept.push("skull")
-      // note : here it's important to set a copy of the array in order for the view to be updated.
-      setDiceKept([...diceKept])
-      const rollDiceWithoutSkulls = removeSkullsFromArray(roll)
-      setDiceRolled([...rollDiceWithoutSkulls])
-    } else {
-      setDiceRolled(roll)
+    rollOnGoingDices(diceOnGoing)
+    if (!diceRolledOnce) setDiceRolledOnce(true)
+
+    autoKeepSkulls()
+
+    if (isGameOver([...diceOnGoing, ...diceKept], currentCard)) {
```

Comment:
isGameOver have to check onGoingDices and keptDices to count how many skulls are visible.
Because even if skulls are auto moved to kept zone it is not done immediatly by react.